src/main/java/org/broadleafcommerce/payment/service/gateway/PayPalSyncTransactionServiceImpl.java

```diff
@@ -0,0 +1,190 @@
+/*-
+ * #%L
+ * BroadleafCommerce PayPal
+ * %%
+ * Copyright (C) 2009 - 2019 Broadleaf Commerce
+ * %%
+ * Licensed under the Broadleaf Fair Use License Agreement, Version 1.0
+ * (the "Fair Use License" located  at http://license.broadleafcommerce.org/fair_use_license-1.0.txt)
+ * unless the restrictions on use therein are violated and require payment to Broadleaf in which case
+ * the Broadleaf End User License Agreement (EULA), Version 1.1
+ * (the "Commercial License" located at http://license.broadleafcommerce.org/commercial_license-1.1.txt)
+ * shall apply.
+ * 
+ * Alternatively, the Commercial License may be replaced with a mutually agreed upon license (the "Custom License")
+ * between you and Broadleaf Commerce. You may not use this file except in compliance with the applicable license.
+ * #L%
+ */
+package org.broadleafcommerce.payment.service.gateway;
+
+import org.apache.commons.lang.StringUtils;
+import org.broadleafcommerce.common.payment.PaymentGatewayType;
+import org.broadleafcommerce.common.payment.PaymentType;
+import org.broadleafcommerce.common.payment.dto.PaymentRequestDTO;
+import org.broadleafcommerce.common.payment.dto.PaymentResponseDTO;
+import org.broadleafcommerce.vendor.paypal.api.ReportingTransactions;
+import org.broadleafcommerce.vendor.paypal.domain.ReportingTransactionResponse;
+import org.broadleafcommerce.vendor.paypal.domain.TransactionDetail;
+import org.broadleafcommerce.vendor.paypal.domain.TransactionInfo;
+import org.broadleafcommerce.vendor.paypal.service.payment.MessageConstants;
+
+import com.paypal.base.rest.APIContext;
+import com.paypal.base.rest.PayPalRESTException;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Resource;
+
+/**
+ * This takes the {@link PaymentRequestDTO} and calls the {@link ReportingTransactions#get(Map, APIContext)} to read all the
+ * {@link ReportingTransactionResponse} and then filters the results by matching the {@link TransactionInfo#getPaypal_reference_id()}
+ * and {@link TransactionInfo#getCustom_field()}
+ *
+ * Note: in the PayPal payload, the {@link TransactionInfo#getCustom_field()} is used to capture and custom info that we want.
+ * This class assumes that a BLC-produced transaction id is being sent to PayPal.
+ *
+ * @author venkat
+ *
+ */
+public class PayPalSyncTransactionServiceImpl implements PayPalSyncTransactionService {
+
+    @Resource(name = "blExternalCallPayPalCheckoutService")
+    protected ExternalCallPayPalCheckoutService payPalCheckoutService;
+
+    @Override
+    public PaymentResponseDTO lookupTransactionByQueryParams(PaymentRequestDTO paymentRequestDTO) throws PayPalRESTException {
```

Comment:
@vpenumatsa12 this class looks like it's on track, but I expected to see a few different things:

1) I'm surprised to not see a reference the the API's usage of pagination. I'd expect the resource calling this service to decide how to handle the paginated lookups, but this class should provide the ability to specify the requested page number.
2) I'm surprised to see the comparison of BillingAgreementIds & transactionIds (customFields) in this class. I had expected this class to simply call the API & return a list of results for the resource calling this service to interpret & filter. That may be closer to the `lookupTransactionsByQueryParams()` method below. It's probably best to move forward with just one of the two public methods.
3) Whichever public method we move forward with, we need to provide javadocs about the expected in-coming query params.